addons/populate/functions/fnc_spawnBuildingUnit.sqf

```diff
@@ -0,0 +1,95 @@
+#include "script_component.hpp"
+/*
+    Author:
+        Bridg
+
+    Description:
+        Spawns a unit at the passed position
+
+    Parameters:
+        0: _spawnPositions <ARRAY>
+        1: _numberOfUnitsToSpawn <NUMBER>
+        2: _numberOfPositionsToOccupy <NUMBER>
+        3: _side <SIDE>
+        4: _module <MODULE>
+        5: _unitPoolArray <ARRAY>
+        6: _statics <ARRAY>
+
+    Return value:
+        Nothing
+*/
+
+params ["_spawnPositions", "_numberOfUnitsToSpawn", "_numberOfPositionsToOccupy", "_side", "_module", "_unitPoolArray", "_statics"];
+
+private _group = grpNull;
+private _currenGrouptUnitCount = 0;
+
+[{
+    params ["_args", "_idPFH"];
+    _args params ["_spawnPositions", "_numberOfUnitsToSpawn", "_numberOfPositionsToOccupy", "_side", "_module", "_unitPoolArray", "_currenGrouptUnitCount", "_group", "_statics"];
+
+    private _pos = [0,0,0];
+    private _dir = -1;
+
+    if (_numberOfPositionsToOccupy <= 1) exitWith {
```

Comment:
If the count of `_statics` is less than `_numberOfPositionsToOccupy` this will never exit as the only subtraction for `_numberOfPositionsToOccupy` is when `_statics` is not empty, requiring it to be equal or greater in length

Reply:
No other exit condition for the pfh with quite a few bits going on is very risky and will likley lead to a runaway endless spawning